index/scorch/snapshot_segment.go

```diff
@@ -248,12 +250,30 @@ func (c *cachedDocs) prepareFields(wantedFields []string, ss *SegmentSnapshot) e
 		}
 		c.m.Lock()
 	}
+
 	c.updateSizeLOCKED()
 
 	c.m.Unlock()
 	return nil
 }
 
+// hasFields returns true if the cache has all the given fields
+func (c *cachedDocs) hasFields(fields []string) bool {
+	c.m.Lock()
+OUTER:
+	for _, field := range fields {
+		for f := range c.cache {
```

Comment:
Rather than iterating the entire map, can't we just do:

```
if _, exists := c.cache[field]; exists {
    continue OUTER
}
```

Reply:
oh, great catch!